source/_posts/2015-01-21-recursion-functions-and-inductive-types.markdown

```diff
@@ -58,11 +58,11 @@ Next, consider the case of lists of natural numbers. Again, we define a function
 
 $$rec_{List \mathbb{N}, X} : X \to (\mathbb{N} \to List \mathbb{N} \to X \to X) \to (List \mathbb{N} \to X)$$
 
-It is easy to generalize these examples to an inductive type $W$ of the form we are considering. Namely, we associate to a constructor type $T$ a recursion data type $R\_X(T)$ as follows:
+It is easy to generalize these examples to an inductive type $W$ of the form we are considering. Namely, we associate to a constructor type $T$ a recursion data type $R\_{W, X}(T)$ as follows:
 
-* if $T = W$, $R\_X(T) = X$
-* if $T = W \to T'$, $R\_X(T) = W \to X \to R\_X(T')$
-* if $T = A \to T'$ with $A$ independent of $W$, then $R\_X(T) = A \to R\_X(T')$
+* if $T = W$, $R\_{W, X}(T) = X$
```

Comment:
Here we define `R_{W, X} ` is function defined on the constructor type `T`. But in the next lecture we define `R_{W, X}` is a function defined on the constructor `\phi`. For the sake of consistency isn't it better to define `R_{W, X}` in terms of constructor here as well?